SalesforceFoundation/HEDAP
# Feature/xinnanli/aff bug

# Critical Changes

# Changes
- We've added a new setting in HEDA Settings, Enable Record Type Validation, to prevent record type errors when users specify an Account as a Contact's Primary Affiliation. When the setting is selected, we now validate that affiliated Accounts have a record type and that all Account Record Type values are correctly mapped to their respective Contact Primary Affl Fields in Affiliation Mappings. If validation fails, an error prevents the Affiliation from being saved. (See <a href="https://powerofus.force.com/articles/Resource/HEDA-Create-Accounts-and-Affiliations">Create Primary Affiliation Mappings</a> to see the correlation between Account record types and Contact Primary Affl fields.) 
- In an existing org, this validation is off by default. To turn it on, you must select Enable Record Type Validation in HEDA Settings, under Affiliations | Settings.

# Issues Closed
Fixes #578
# New Metadata

# Deleted Metadata

# Testing Notes
1. Create accounts without record types before deploying HEDA
2. Deploy HEDA
3. Create a contact and use one of the testing account as primary field (Business org etc).
4. Go to HEDA settings and check Affiliation Record Type enforced setting
4. Change the primary field to another account without record type/ uncheck primary checkbox on the new affiliation created.
5. Make sure the proper error message is shown.
6. Create another contact and use another account as primary field (Business or etc). 
7. Go to setting page and modify the affiliation account mappings to not matching existing record types.
8. Change the primary field to another account without record type/ uncheck primary checkbox on the new affiliation created.
9. Make sure the proper error message is shown.
